FaynaTown.Tennis.Planner.Bot.Core/Services/ReservationService.cs

```diff
@@ -128,7 +155,7 @@ internal sealed class ReservationService : IReservationService {
     }
 
     public bool CancelReservation(Reservation reservationToCancel) {
-        if (Reservations.Value.Any(reservation => reservation == reservationToCancel)) {
+        if (Reservations.Value.Any(reservation => reservation.Equals(reservationToCancel))) {
```

Comment:
```suggestion
        if (Reservations.Value.Contains(reservationToCancel)) {
```
It looks simpler with `.Contains()`.
`Reservation` should implement `IEquatable<T>`, however. But It's almost implemented anyway.
`IEquatable<T>.Equals` also 

> [offers slightly better performance](https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1.equals?view=net-8.0#remarks:~:text=Of%20the%20two%20methods%2C%20IEquatable%3CT%3E.Equals%20offers%20slightly%20better%20performance)

If this change is to be applied `User` should implement `IEquatable<T>` too to be consistent.